### DISCUSSION TITLE
Part images not shown

### DISCUSSION THREAD

**Discussion by SpArX0110** [inventree/InvenTree#4395] [Q&A] [answered]
Hello,

I am trying to upload images for parts, but after the upload they are not shown.
Is there something in the settings that I have to do and I just can't find it? Or do the pictures have to be a specific format?

<img width="936" alt="Screenshot 2023-02-23 at 09 33 13" src="https://user-images.githubusercontent.com/84473777/220859024-6746c5b2-7329-428d-a7d1-a7ac37594de7.png">

**Comment by wolflu05** [comment]
Hello @SpArX0110 ,

can you share the outputs of the developer tools (console and network tab). You can open them via <kbd>STRG</kbd>+<kbd>SHIFT</kbd>+<kbd>I</kbd> or <kbd>F12</kbd> . There should appear some errors why the pictures cannot be loaded.

**Reply by SpArX0110** [reply]
Thanks for the reply, but we fixed it :)

**Comment by jpeterse** [comment]
I just upgraded to 1.0.5, and I am running into the same issue. @SpArX0110 What was your solution to the problem?

For me, In dev console. I see the response for images return a 302 code. When I look at the network traffic and look for the response, I see a text error "If you see this text there might be an Issue with your update - see [INVE-E1](https://docs.inventree.org/en/stable/settings/error_codes/#inve-e1) in the docs for help".

I run the packaged version of Inventree, and have run "inventree run cli update" to try and resolve, but to no avail.
I have also checked the path in config.yaml is correct, and that the user (inventree in my case) have access to the image files. I can list all the files in parts_image directory.

**Comment by SchrodingersGat** [comment]
@SpArX0110 if you can share your solution here, that would be helpful. Perhaps we need to adjust the docs?

**Comment by jpeterse** [comment]
To provide more context.
1. Opened dev console to see details as reported in the browser
<img width="1648" height="577" alt="inventree_part_images_issue_1" src="https://github.com/user-attachments/assets/9be126fc-8426-4eae-a6bb-58fcbae104f5" />

2.  Confirmed the user that Inventree is running as, and that file access is working for that user.
<img width="1337" height="274" alt="inventree_part_images_issue_2" src="https://github.com/user-attachments/assets/0f02f506-dc5c-4a71-b598-2f275adfb7eb" />

3. Enabled Debug mode, allowing Inventree to handle static and media files manually. 
**Now the part images appear.**

5. Compared my NGINX configuration file with the one provided at [here](https://github.com/inventree/inventree/blob/466463ad74ae93fa61d2a1215773db6b561907cb/contrib/container/nginx.conf)

`root@dev:/tmp# diff /tmp/nginx.conf /etc/nginx/sites-available/inventree.conf 
1,3d0
< # An example configuration file for running InvenTree container behind an nginx proxy
< # While suitable for a simple installation, this file will likely require some modification
< # if you are running a more complex setup (e.g behind another proxy, or with HTTPS)
root@dev:/tmp# 
`

Only difference is the introduction statement in the sample file.
I don't use ssl or anything specific. Only thing I can think of that might be different from standard config, is that my Inventree instance run on a remote host, not the local machine. Thus in the Inventree configuration file, I changed the site_url configuration parameter and in conf.d/other I changed the environment variable to match the url (replacing "localhost" width "dev.home.arpa". 

Why running _apt upgrade_ would break the setup still eludes me. Inventree was working fine before the upgrade.

**Reply by SchrodingersGat** [reply]
@SpArX0110 likely the host name does not match correctly which is causing the `INVE-E1` errors.

You should look at the [server access documentation](https://docs.inventree.org/en/stable/start/config/#server-access) - in particular the `INVENTREE_TRUSTED_ORIGINS` - you might need to add your access URL here.


> Why running apt upgrade would break the setup still eludes me. Inventree was working fine before the upgrade.

What version did you update *from*? In the 1.0.0 release there were some large breaking changes regarding security which came from upstream django updates

**Comment by jpeterse** [comment]
Ahh, that could be the reason. Just checked, and I updated from 0.17.7.

fqdn of the host:
```
root@dev:/etc/inventree# hostname -A
dev.home.arpa 
```
The parts I have in config.yaml now that reference the fqdn are:
```
# Base URL for the InvenTree server (or use the environment variable INVENTREE_SITE_URL)
site_url: 'http://dev.home.arpa:8000'
```
and I just now added
```
trusted_origins:
   - 'http://localhost'
   - 'http://*.localhost'
   - 'http://*.home.arpa'
```
Haven't resolved the issue yet.
I already had the INVENTREE_SITE_URL set, which I understood to then automatically take care of CORS and trusted CSRF ?

**Reply by matmair** [reply]
Only if you are not overriding the values

**Reply by jpeterse** [reply]
I'm not educated to understand what you mean, in that I don't know what would overwrite the values?
(I'm not a developer. I don't know anything about django etc. That's why I use the packaged version of Inventree)
Only place I found that also had it set was in the conf.d/other environment file. I already had it changed as well to:

```
root@dev:/etc/inventree# cat conf.d/other 
export INVENTREE_CONFIG_FILE="/etc/inventree/config.yaml"
#export INVENTREE_SITE_URL="http://47.189.139.243"
export PORT="8000"
export INVENTREE_SITE_URL="http://dev.home.arpa:8000"
```
Are the any other places where the urls could be overwritten?

**Comment by jpeterse** [comment, marked as answer]
Issue resolved. Part user error, part configuration change.

User error:
I called http://dev.home.arpa:8000 as the url in the browser. I.e. I bypassed the NGINX serves, so wasn't served static and media files at all.

configuration:
I had to update the NGINX configuration.
 starting with the [sample](https://github.com/inventree/inventree/blob/466463ad74ae93fa61d2a1215773db6b561907cb/contrib/container/nginx.conf) NGINX configuration file 
  -    I added a server_name directive before the first location block ( "server_name inventree.home.arpa" in my case)
  -    I updated the 2 proxy_pass lines, replacing "http://inventree-server:8000" with "http://localhost:8000"
  

### In summary
4 configuration changes were needed, with an optional 5th:
1. Update config.yaml to set _site_url_ to the hostname of the server
2. Update config.yaml enabling _trusted_domains_ (still not quite sure this really is needed, but hey)
3. Update conf.d/other fixing the export statement _INVENTREE_SITE_URL_ to have the same setting as in the config.yaml (don't really sure why it's overwritten here, when it's also specified in the config.yaml)
4. Update NGINX config in _sites-enabled_ folder, change 2 instances of _proxy_pass_ directive to point to localhost instead of the docker standard "inventree-server".
5. (Optional) Update NGINX config to add a server_name directive, to allow the NGINX server to serve other applications as well.

AND url to access the server is just the hostname of the server.
